src/Sulu/Component/Rest/ListBuilder/Expression/Doctrine/AbstractDoctrineExpression.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Rest\ListBuilder\Expression\Doctrine;
+
+use Doctrine\ORM\QueryBuilder;
+use Sulu\Component\Rest\ListBuilder\Expression\ExpressionInterface;
+
+/**
+ * Abstract definition for expressions used by the DoctrineListbuilder.
+ */
+abstract class AbstractDoctrineExpression implements ExpressionInterface
+{
+    /**
+     * Returns a statement for an expression.
+     *
+     * @param QueryBuilder $queryBuilder
+     *
+     * @return string
+     */
+    abstract public function getStatement(QueryBuilder $queryBuilder);
```

Comment:
I am not really sure what we will need for the implementation for PHPCR and thats the reason I did it this way. When I implement the filters for the snippets we will se what can be abstracted and what not. I would leave it this way for now.